### DISCUSSION TITLE
How can I repeat a shell command after "scroll_to_prompt"

### DISCUSSION THREAD

**Discussion by seh** [kovidgoyal/kitty#6514] [Q&A]
If I press the <kbd>C-S-z</kbd> keys to invoke `scroll_to_prompt -1` one or more times and my cursor lands on a prompt with that command that I'd like to either invoke as is or edit before invoking, are there any key bindings available to capture that command? If not, is there a key binding I can write myself that would copy that command and paste it into a fresh prompt?

**Comment by kovidgoyal** [comment]
I dont think there is such an action, someone would have to code it. But
why not use shell history for this?

**Reply by seh** [reply]
I do normally use <kbd>C-r</kbd> in concert with _fzf_ to do this sort of thing, but I've been experimenting with _kitty_'s ability to move around through the scrollback buffer, and this just felt like a natural thing to want to do, since I'm staring at a command that I decide that I want to repeat or edit slightly before trying again. Seeing it "right there" but not being able to act on it feels limiting.

**Reply by kovidgoyal** [reply]
well scroll to prompt is designed mostly for getting access to the *output* of previous commands not the commands themselves, but it should be easy enough to add a function to get the output of the command itself, PR welcome.
